sdk_extensions/aws_v1_support/build.gradle

```diff
@@ -16,6 +16,11 @@ dependencies {
             libraries.guava,
             platform(boms.guava)
 
+    implementation 'com.fasterxml.jackson.core:jackson-databind:2.11.0',
+            libraries.guava,
+            platform(boms.guava)
+    implementation 'io.kubernetes:client-java:9.0.0'
```

Comment:
This is a pretty heavyweight dependency, just one of the transitive dependencies alone is 8MB and I think total it will possibly double the agent size.

Can you try making direct calls to the API? FWIU, token is just a file to read so authentication is very easy for k8s API so it's mostly just parsing, which doesn't seem too bad.

Reply:
use HttpURLConnection instead of k8s client